Sources/NIOHTTPTypesHTTP1/HTTPTypeConversion.swift

```diff
@@ -15,14 +15,14 @@
 import HTTPTypes
 import NIOHTTP1
 
-private enum HTTP1TypeConversionError: Error {
+public enum HTTP1TypeConversionError: Error {
```

Comment:
We have to do our usual struct wrapping enum dance here otherwise we can never evolve this enum in the future.

Reply:
Done.
I was wondering why NIO never implemented a `func ~` for these errors so switch and catch pattern matching was simplified. Currently to catch one of these "struct" errors you need
```swift
} catch let e as HTTP1TypeConversionError where e == .invalidStatusCode {
```
Which is a mouthful, and something I get wrong every time.
If this was added to allow pattern matching
```swift
    public static func ~= (_ lhs: Self, _ rhs: Error) -> Bool {
        guard let rhs = rhs as? Self else { return false }
        return lhs == rhs
    }
```
The catch statement gets reduced to
```swift
} catch HTTP1TypeConversionError.invalidStatusCode {
```
as if it is an enum.

Reply:
In general, we try to stay away from custom operators or overloading more operators since they have quite a heavy impact on the compiler's performance. What we really want is extensible enums in non-library evolution mode.

Reply:
> What we really want is extensible enums in non-library evolution mode.

Been waiting for that one for a while.

Reply:
@FranzBusch Is there anything holding merging this now?